selfapi.js

```diff
@@ -405,21 +430,30 @@ API.prototype = {
       html += '<h2 id="' + getAnchor(title) + '">' + title + '</h2>\n';
       html += '<pre>' + method.toUpperCase() + ' ' + fullPath + '</pre>\n';
       if (handler.description) {
-        html += '<p>' + handler.description + '</p>\n';
+        html += '<p>' + handler.description.replace(/\n/g, '<br/>') + '</p>\n';
       }
 
       if (handler.examples && handler.examples.length > 0) {
         var example = handler.examples[0];
 
         var request = example.request || {};
-        if (request.headers || request.body) {
+        if (request.headers || request.body || request.urlParameters) {
           html += '<h3>Input</h3>\n<pre>';
+
+          // Format example-specific URL
+          var exampleURL = method.toUpperCase() + ' ' + fullPath;
+          var requestParameters = request.urlParameters || {};
+          for (var requestParam in requestParameters) {
+            exampleURL = exampleURL.replace(':' + requestParam, requestParameters[requestParam]);
+          }
+          html += exampleURL + '\n';
+
           var requestHeaders = request.headers || {};
           for (var header in requestHeaders) {
             html += header + ': ' + requestHeaders[header] + '\n';
           }
           if (request.body) {
-            html += '\n' + request.body.trim() + '\n';
+            html += '\n' + ensureSerialized(request.body, 4).trim() + '\n';
```

Comment:
Nit: Why enforce a default JSON spacing of 4 here? I think it would be less surprising for API implementors if their own conventions are used (e.g. my suggested `exports.options.jsonStringifySpaces` preference, if it's specified).

Additionally, if you agree that JSON outputs should be made human-readable by default (as I do), I would suggest using a default spacing of 2 if nothing is specified by the implementor. This makes every API human-readable by default, respecting (maybe) the most common JS indentation, while virtually adding no overhead to HTTP transfer sizes (given that gzip produces approximately the same output size for single-line JSON and human-readable JSON).